### PR TITLE
Replace array.at(-1) with array[array.length - 1]

### LINKED ISSUE #6831: `useOverlayStack()` - `Array.at` incompatible with older browsers (Safari 15.3, older)
#### Environment

- __Package version(s)__: @blueprintjs/core 5.10.2, develop
- __Operating System__: Mac OS Big Sur
- __Browser name and version__: Safari <15.4

#### Steps to reproduce

1. Open any Overlay2 in Safari 15.3 or earlier

#### Actual behavior

```TypeError stack.current.at is not a function. (In 'stack.current.at(-1)', 'stack.current..at' is undefined)``` 

which refers to https://github.com/palantir/blueprint/blob/develop/packages/core/src/hooks/overlays/useOverlayStack.ts#L73 (also https://github.com/palantir/blueprint/blob/develop/packages/core/src/hooks/overlays/useLegacyOverlayStack.ts#L65)

#### Expected behavior

Safari 15.3 should be supported, it's only 2 years old (https://blueprintjs.com/docs/#blueprint/principles.browser-support)

#### Possible solution

Replace `stack.current.at(-1)` with `stack.current[stack.current.length - 1]` - `Array.at` is a relatively recent addition (https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/at) and is not used elsewhere in the Blueprint codebase. It seems like an easy change to make to improve compatibility with older browsers.

### PR DESCRIPTION
**Pull request by jonaswitt**
#### Checklist

- [ ] Includes tests (n/a?)
- [ ] Update documentation (n/a?)

#### Changes proposed in this pull request:

See #6831

#### Reviewers should focus on:

Testing on Safari 15.3 or earlier

### CODE REVIEW

**Comment by palantirtech** [comment]
Thanks for your interest in [palantir/blueprint](https://github.com/palantir/blueprint), @jonaswitt! Before we can accept your pull request, you need to sign our contributor license agreement - just visit https://cla.palantir.com/ and follow the instructions. Once you sign, I'll automatically update this pull request.

**Comment by jonaswitt** [comment]
Not sure what to do about the CircleCI Pipeline errors, sorry. It's my first PR in this repo.

**Comment by gluxon** [comment]
> Not sure what to do about the CircleCI Pipeline errors, sorry. It's my first PR in this repo.

I'm confused by them too actually.

Could you push another commit to see if it's still happening? The new commit can be empty:

```sh
git commit --allow-empty -m 'Empty commit to rerun CircleCI pipeline'
```

File: packages/core/src/hooks/overlays/useLegacyOverlayStack.ts [hunk 1]
```diff
@@ -62,7 +62,7 @@ export function useLegacyOverlayStack(): UseOverlayStackReturnValue {
         legacyGlobalOverlayStackStore.getSnapshot,
     );
 
-    const getLastOpened = React.useCallback(() => stack.at(-1), [stack]);
+    const getLastOpened = React.useCallback(() => stack.at(stack.length - 1), [stack]);
```
Comment (gluxon) [inline packages/core/src/hooks/overlays/useLegacyOverlayStack.ts]: I think we need to remove `.at` entirely to fix the browser compatibility concern?

```suggestion
    const getLastOpened = React.useCallback(() => stack[stack.length - 1], [stack]);
```

File: packages/core/src/hooks/overlays/useOverlayStack.ts [hunk 2]
```diff
@@ -70,7 +70,7 @@ export function useOverlayStack(): UseOverlayStackReturnValue {
     const legacyOverlayStack = useLegacyOverlayStack();
 
     const getLastOpened = React.useCallback(() => {
-        return stack.current.at(-1);
+        return stack.current.at(stack.current.length - 1);
```
Comment (gluxon) [inline packages/core/src/hooks/overlays/useOverlayStack.ts]:
```suggestion
        return stack.current[stack.current.length - 1];
```

**Comment by gluxon** [comment]
Hm.. looks like the empty commit isn't working. Thanks for trying that though.

It looks like CircleCI uses your GitHub account, which surprises me. I would have expected it to have access to the repo directly. Sorry about that. Could you try the instructions here?

https://support.circleci.com/hc/en-us/articles/12504745606171-Why-am-I-seeing-the-Could-not-find-a-usable-config-yml-you-may-have-revoked-the-CircleCI-OAuth-app-message

File: packages/core/src/hooks/overlays/useLegacyOverlayStack.ts (hunk 1 above)
Comment (jonaswitt) [reply packages/core/src/hooks/overlays/useLegacyOverlayStack.ts]: I think you're right! Thanks for the feedback.

Comment (gluxon) [reply packages/core/src/hooks/overlays/useLegacyOverlayStack.ts]: Np! Thanks for the fix.

**Review by gluxon — APPROVED** [review]
Approving. Thanks for the fix @jonaswitt! I'll merge once CI passes.

**Pull request merged**